polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala

```diff
@@ -140,17 +100,21 @@ class PythonInterpreter(val symbolTable: RuntimeSymbolTable) extends LanguageInt
   private val shutdownSignal = ReadySignal()(listenerShift)
 
   override def runCode(
-    cell: CellID,
-    visibleSymbols: Seq[Decl],
-    previousCells: Seq[CellID],
+    cellContext: CellContext,
     code: String
   ): IO[Stream[IO, Result]] = if (code.trim().isEmpty) IO.pure(Stream.empty) else {
     val run = new global.Run()
+
+    val cell = cellContext.id
     val cellName = s"Cell$cell"
     global.newCompilationUnit("", cellName)
 
     val shiftEffect = IO.ioConcurrentEffect(shift) // TODO: we can also create an implicit shift instead of doing this, which is better?
     Queue.unbounded[IO, Option[Result]](shiftEffect).flatMap { maybeResultQ =>
+      // TODO: Jep doesn't give us a good way to construct a dict... should we wrap cell context in a class that emulates dict instead?
```

Comment:
We use the visible values from the `CellContext` as the globals dict. It just needs a bit of finagling to get through Jep in a usable way.

Reply:
Hmm, actually I might have forgotten to also do this in `completionsAt`.

Reply:
I guess their PyJMap conversion is not good enough because it doesn't support `__iter__()`?  If so, seems like an oversight... https://github.com/ninia/jep/blob/master/release_notes/3.4-notes.rst#added-pyjmap-type